monai/data/utils.py

```diff
@@ -50,8 +50,11 @@
     issequenceiterable,
     look_up_option,
     optional_import,
+    pytorch_after,
 )
 
+if pytorch_after(1, 13):
+    from torch.utils.data._utils.collate import collate_tensor_fn, default_collate_fn_map
```

Comment:
Since the module is `_utils` I wonder if this is meant to be private code we shouldn't rely on. What is the advantage of using these functions over the ones we've been using so far?

Reply:
The benefits may be we could simplify L463-L466 after we add a new collate map for `MetaTensor`.
https://github.com/Project-MONAI/MONAI/blob/85243f52085b0946870fc7808c484c8b639eaf1b/monai/data/utils.py#L463-L466

Reply:
I would just be careful using `_utils` directly. [Here](https://github.com/pytorch/pytorch/blob/d7a3f2128fb4457dd60fd5d23e77d2c66a8b0f02/torch/utils/data/dataloader.py#L64C37-L64C37) the function is imported into a public module. It seems `torch.utils.data.dataloader.default_collate` is meant to be the public member which just happens to be implemented as an alias to a private one, but this could change without breaking the public interface.

Reply:
Yes, we should always be careful with the private code. I didn't find an alias for these two. 
But it seems they tend to extend collate functions that can register collate functions to handle specific types.
https://github.com/pytorch/pytorch/pull/85748

Do you have some suggestions for using these or we could just leave them here?

Reply:
It seems that `default_collate_fn_map` is meant to be internal and used only by `default_collate`, but `collate` is also private so shouldn't be used either? We have no way of tweaking collate behaviour if so. I wouldn't myself play around with Pytorch internals like this but there's no other way it seems to get your behaviour and reuse things. Let's put in comments where you do the imports stating you're importing private code for reuse purposes so we keep track in case things break in the future.

Reply:
Sure, thanks! I added comments in the latest commit.